nomos-services/mempool/src/tx/service.rs

```diff
@@ -6,28 +6,71 @@ pub mod openapi {
 
 // std
 use std::fmt::Debug;
+use std::marker::PhantomData;
+use std::path::PathBuf;
 
 // crates
 use futures::StreamExt;
+use overwatch_rs::services::state::ServiceState;
+use serde::de::DeserializeOwned;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+use tracing::error;
+use utils::overwatch::recovery::backends::FileBackendSettings;
+use utils::overwatch::{JsonFileBackend, RecoveryOperator};
 // internal
-use crate::backend::MemPool;
+use crate::backend::{MemPool, RecoverableMempool};
 use crate::network::NetworkAdapter;
 use crate::{MempoolMetrics, MempoolMsg};
 use nomos_network::{NetworkMsg, NetworkService};
 use overwatch_rs::services::life_cycle::LifecycleMessage;
 use overwatch_rs::services::{
     handle::ServiceStateHandle,
     relay::{OutboundRelay, Relay},
-    state::{NoOperator, NoState},
     ServiceCore, ServiceData, ServiceId,
 };
-use tracing::error;
+
+#[derive(Error, Debug)]
+pub enum StateError {}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct MempoolState<NS, PS, P> {
+    // `pool` is an option so that we don't need it to implement `Mempool`
+    // when implementing `ServiceState` for this type, as otherwise we would need to call `P::new(settings)`.
```

Comment:
There's another nuance to this: You need the attributes to be `Option`s as that's what provides the ability to differentiate whether there was a previously saved value or there was none.
I would suggest updating the comment (or maybe even removing it?) so as to avoid confusion with this behaviour. Or do you feel it makes sense as it is?

Reply:
Yes, I will rework this whole PR actually once I merge the two outstanding ones in the overwatch repo, as I feel that is enough of a refactor to justify rewriting the few lines I wrote in this PR. But yours is a good point, and I will change that as well.